services/src/unit-test/java/uk/gov/moj/sdt/services/TargetApplicationSubmissionServiceTest.java

```diff
@@ -307,225 +394,212 @@ public Object answer() throws Throwable {
                 receiveTimeOutParam);
 
         final TimeoutException timeoutEx = new TimeoutException("Timeout occurred", "Timeout occurred");
         this.mockConsumerGateway.individualRequest(individualRequest, 1000, 12000);
-        EasyMock.expectLastCall().andThrow(timeoutEx);
+        doThrow(timeoutEx).when(mockConsumerGateway).individualRequest(individualRequest, 1000, 12000);
 
         final IErrorMessage errorMsg = new ErrorMessage();
         errorMsg.setErrorCode("REQ_NOT_ACK");
         errorMsg.setErrorDescription("Request not acknowledged");
         errorMsg.setErrorText("Request Not Acknowledged");
 
-        EasyMock.expect(this.mockErrorMsgCacheable.getValue(IErrorMessage.class, "REQ_NOT_ACK")).andReturn(errorMsg);
+        when(this.mockErrorMsgCacheable.getValue(IErrorMessage.class, "REQ_NOT_ACK")).thenReturn(errorMsg);
 
         // Now create an ErrorLog object with the ErrorMessage object and the IndividualRequest object
         final IErrorLog errorLog = new ErrorLog(errorMsg.getErrorCode(), errorMsg.getErrorText());
 
         individualRequest.setErrorLog(errorLog);
 
-        this.mockIndividualRequestDao.persist(individualRequest);
-        EasyMock.expectLastCall();
-
         final IGlobalParameter maxForwardingAttemptsParam = new GlobalParameter();
         maxForwardingAttemptsParam.setName("MAX_FORWARDING_ATTEMPTS");
         maxForwardingAttemptsParam.setValue("3");
-        EasyMock.expect(this.mockCacheable.getValue(IGlobalParameter.class, "MAX_FORWARDING_ATTEMPTS")).andReturn(
+        when(this.mockCacheable.getValue(IGlobalParameter.class, "MAX_FORWARDING_ATTEMPTS")).thenReturn(
                 maxForwardingAttemptsParam);
 
-        this.mockMessageWriter.queueMessage(EasyMock.isA(ISdtMessage.class), EasyMock.isA(String.class),
-                EasyMock.anyBoolean());
-        EasyMock.expectLastCall();
+        this.mockMessageWriter.queueMessage(isA(ISdtMessage.class), isA(String.class), anyBoolean());
 
-        EasyMock.replay(mockIndividualRequestDao);
-        EasyMock.replay(mockConsumerGateway);
-        EasyMock.replay(mockCacheable);
-        EasyMock.replay(mockMessageWriter);
-        EasyMock.replay(mockErrorMsgCacheable);
+        // Create appender and add to logger
+        ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+        listAppender.start();
+        logger.addAppender(listAppender);
 
         this.targetAppSubmissionService.processRequestToSubmit(sdtRequestRef);
 
-        EasyMock.verify(mockIndividualRequestDao);
-        EasyMock.verify(mockConsumerGateway);
-        EasyMock.verify(mockCacheable);
-        EasyMock.verify(mockMessageWriter);
-        EasyMock.verify(mockErrorMsgCacheable);
+        List<ILoggingEvent> logList = listAppender.list;
+
+        assertFalse(logList.isEmpty());
+
+        logger.detachAndStopAllAppenders();
+
+        verify(mockIndividualRequestDao,times(2)).persist(individualRequest);
```

Comment:
Add code to verify behaviour of other mocked objects.

Reply:
Done